code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplateBroadcastReceiver.java

```diff
@@ -0,0 +1,302 @@
+            handleFilmstripNotificationIntent(context, intent);
+        }
+    }
+
+    private void handleFilmstripNotificationIntent(final Context context, final Intent intent) {
+        final NotificationManagerCompat notificationManager =
+                NotificationManagerCompat.from(context);
+        final String packageName =
+                ServiceProvider.getInstance()
+                        .getAppContextService()
+                        .getApplication()
+                        .getPackageName();
+
+        // get filmstrip notification values from the intent extras
+        final Bundle intentExtras = intent.getExtras();
+        final String backgroundColorHex =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.BACKGROUND_COLOR);
+        final String titleColorHex =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.TITLE_COLOR);
+        final String bodyColorHex =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.BODY_COLOR);
+        final String iconColorHex =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.ICON_COLOR);
+        final String titleText =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.SMALL_TITLE_TEXT);
+        final String smallBodyText =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.SMALL_BODY_TEXT);
+        final String expandedBodyText =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.EXPANDED_BODY_TEXT);
+        final String leftImageUri =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.LEFT_IMAGE);
+        final String centerImageUri =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.CENTER_IMAGE);
+        final String rightImageUri =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.RIGHT_IMAGE);
+        final String leftCaptionText =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.LEFT_CAPTION);
+        final String centerCaptionText =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.CENTER_CAPTION);
+        final String rightCaptionText =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.RIGHT_CAPTION);
+        final String channelId =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.CHANNEL_ID);
+        final String messageId =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.MESSAGE_ID);
+
+        final Map<String, String> captionMap =
+                new HashMap<String, String>() {
+                    {
+                        put(leftImageUri, leftCaptionText);
+                        put(centerImageUri, centerCaptionText);
+                        put(rightImageUri, rightCaptionText);
+                    }
+                };
```

Comment:
See my comment on handling more than 3 images. If we pursue that, we will not need these extras.